Bottom: 831be5899e7800a691794b416b4902d33a6b735d
Top:    bcdda0a726d6f0264aba4f277ecddef002af1b0a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-03 00:24:03 +0100

Refresh of crosshair-c-revise-heuristic-f

---

diff --git a/src/crosshair.c b/src/crosshair.c
index b42afa9..bb0a87b 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -846,16 +846,17 @@ check_snap_offgrid_line (struct snap_data *snap_data,
 
   line = (LineType *)ptr2;
 
-  /* Allow snapping to off-grid lines when drawing new lines,
-   * when moving a line end-point (but don't snap to the same line)
+  /* Allow snapping to off-grid lines when drawing new lines (on
+   * the same layer), and when moving a line end-point
+   * (but don't snap to the same line)
    */
-#if 0
-  if ( Settings.Mode != LINE_MODE &&
+  if ((Settings.Mode != LINE_MODE ||
+       Crosshair.AttachedObject.Ptr1 != ptr1) &&
       (Settings.Mode != MOVE_MODE ||
+       Crosshair.AttachedObject.Ptr1 != ptr1 ||
        Crosshair.AttachedObject.Type != LINEPOINT_TYPE ||
        Crosshair.AttachedObject.Ptr2 == line))
     return;
-#endif
 
   dx = line->Point2.X - line->Point1.X;
   dy = line->Point2.Y - line->Point1.Y;
